tests/luks/test_luks_cold_migration.py

```
@@ -0,0 +1,332 @@
+
+        vms = [copy.deepcopy(vm) for vm in prepared_plan["virtual_machines"]]
+        for vm in vms:
+            vm.pop("luks", None)
+            luks_passphrase = vm.pop("luks_passphrase", None)
+            if luks_passphrase is None:
+                luks_passphrase = source_provider_data.get("luks_passphrase")
+            if not luks_passphrase:
+                raise ValueError(
+                    f"LUKS passphrase not found for VM '{vm['name']}' — "
+                    "checked per-VM config and source_provider_data['luks_passphrase']"
+                )
```

Ruff (0.15.17)

[warning] Avoid specifying long messages outside the exception class

(TRY003)